_pages/dev/components/shared-components/table-component.md

```diff
@@ -0,0 +1,140 @@
+
+The following is a basic example of a table configuration, which provides the cell definition for the "budget" table:
+
+```ts
+provideConfig({
+  table: {
+    budget: {
+      cells: ['name', 'active', 'amount', 'dateRange', 'unit'],
+    }
+  }
+} as TableConfig)
+```
+
+## Table Layout Configuration
+
+The table can be rendered in the following different layout orientations:
+
+- `VERTICAL` renders the table vertically, with a heading on top of the table
+- `VERTICAL_STACKED` renders a horizontally-oriented table layout, with the table headers in the first column of the table
+- `HORIZONTAL` renders the table horizontally, with the header on the left side of the table
+
+The vertical layout is the default layout.
+
+A layout configuration can be provided for each table type, as shown in the following example:
+
+```ts
+provideConfig({
+  table: {
+    budget: {
+      cells: ['name', 'active', 'amount', 'dateRange', 'unit'],
+      options: {
+        layout: TableLayout.VERTICAL,
+      }
+    }
+  }
+} as TableConfig)
+```
+
+## Responsive Table Configuration
+
+Table configurations can be further configured for each breakpoint, to change the table for each screen size. A mobile-first approach is used, and cascades to larger screen size(s) to keep the configuration as simple as possible.
+
+The following example demonstrates building a configuration on top of the preceding configuration. First, `my-table` is defined, and then it is redefined as `md` for medium-sized (tablet) screens:
+
+```ts
+provideConfig({
+  table: {
+    'budget': {
+      cells: ['name', 'active', 'amount', 'dateRange', 'unit']
+      options: {
+        layout: TableLayout.VERTICAL
+      },
+      md: {
+      cells: ['name', 'unit']
+        options: {
+          layout: TableLayout.HORIZONTAL
+        }
+      }
+    }
+  }
+} as TableConfig)
+```
+
+## Cell Renderers
+
+One of the key features of the table component is the ability to provide (custom) cell components for cell rendering. While a default component is provided for both header and data cells, custom components can be configured to change the rendering. You can configure the rendering of headers and data cells separately.
+
+The following example demonstrates how cell rendering can be configured:
+
+```ts
+provideConfig({
+  table: {
+    'budget': {
+      cells: ['name', 'active', 'amount', 'dateRange', 'unit']
+      options: {
+        headerComponent: BudgetHeaderCellComponent,
+        cells: {
+          name: {
+            dataComponent: BudgetNameDataCellComponent,
+          },
+          active: {
+            headerComponent: BudgetActiveHeaderComponent,
+          }
+        }
+      }
+    }
+  }
+} as TableConfig)
+```
+
+## Cell Outlets
+
+Aside from configuration, the table component can be customized with outlets. Outlets are used to customize existing UI elements without changing existing component logic or templates.
```

Comment:
I would not mind a link to outlets here as well, as soon as we introduce the topic.

Reply:
Done 🙂